base/gmp.jl

```diff
@@ -907,70 +909,119 @@ end
 function Rational{BigInt}(num::BigInt, den::BigInt)
     if iszero(den)
         iszero(num) && __throw_rational_argerror_zero(BigInt)
-        num = isneg(num) ? -one(BigInt) : one(BigInt)
-        return unsafe_rational(BigInt, num, den)
+        return set_si!(z, flipsign(1, num), 0)
```

Comment:
```suggestion
        return set_si(flipsign(1, num), 0)
```
There is a problem here otherwise, since `z` is not defined. Could you check that this code path is tested somewhere by the way?